fdbclient/SpecialKeySpace.actor.cpp

```diff
@@ -1263,3 +1270,53 @@ Future<Optional<std::string>> ConsistencyCheckImpl::commit(ReadYourWritesTransac
 	                          BinaryWriter::toValue(entry.present(), Unversioned()));
 	return Optional<std::string>();
 }
+
+TracingOptionsImpl::TracingOptionsImpl(KeyRangeRef kr) : SpecialKeyRangeRWImpl(kr) {
+	TraceEvent("TracingOptionsImpl::TracingOptionsImpl").detail("Range", kr);
+}
+
+Future<Standalone<RangeResultRef>> TracingOptionsImpl::getRange(ReadYourWritesTransaction* ryw,
+                                                                KeyRangeRef kr) const {
+	Standalone<RangeResultRef> result;
+	if (kr.contains(getKeyRange().begin.withSuffix(kTracingTransactionIdKey))) {
```

Comment:
I think this can return keys out of order, since "transaction_id" > "token"

Reply:
Oh good catch. I could swap the statements, but that doesn't feel like a very future-proof solution. What do you think about doing something like this:

https://github.com/apple/foundationdb/blob/b69d0fa68e0d1b7b20e971cfbf3a66d21aa60132/fdbclient/SpecialKeySpace.actor.cpp#L593

where you loop over the keys in lexicographic order and have an if/switch statement to handle each key?

Reply:
That's a nice way to do it. I like it